internal/service/wallet.go

```diff
@@ -984,3 +999,267 @@ func (s *walletService) GetCustomerWallets(ctx context.Context, req *dto.GetCust
+			"current_balance", currentBalance,
+			"credit_balance", creditBalance,
+			"alert_state", w.AlertState,
+		)
+
+		// If current state is alert, update to ok (recovery)
+		if w.AlertState == string(types.AlertStateAlert) {
+			if err := s.UpdateWalletAlertState(ctx, w.ID, types.AlertStateOk); err != nil {
+				s.Logger.Errorw("failed to update wallet alert state",
+					"wallet_id", w.ID,
+					"error", err,
+				)
+				return err
+			}
+			s.Logger.Infow("wallet recovered from alert state",
+				"wallet_id", w.ID,
+			)
+			return s.PublishEvent(ctx, types.WebhookEventWalletUpdated, w)
+		}
+		return nil
+	}
+
+	// Skip if already in alert state
+	if w.AlertState == string(types.AlertStateAlert) {
+		s.Logger.Infow("skipping alert - already in alert state",
+			"wallet_id", w.ID,
+		)
+		return nil
+	}
+
+	s.Logger.Infow("balance below/equal threshold - triggering alert",
+		"wallet_id", w.ID,
+		"threshold", threshold,
+		"current_balance", currentBalance,
+		"credit_balance", creditBalance,
+	)
+
+	// Update wallet state to alert
+	if err := s.UpdateWalletAlertState(ctx, w.ID, types.AlertStateAlert); err != nil {
+		s.Logger.Errorw("failed to update wallet alert state",
+			"wallet_id", w.ID,
+			"error", err,
+		)
+		return err
+	}
+
+	// Trigger alerts based on which balance is below threshold
+	var errs []error
+	if isCreditBalanceBelowThreshold {
+		s.Logger.Infow("triggering credit balance alert",
+			"wallet_id", w.ID,
+			"credit_balance", creditBalance,
+			"threshold", threshold,
+		)
+		if err := s.PublishEvent(ctx, types.WebhookEventWalletCreditBalanceDropped, w); err != nil {
+			s.Logger.Errorw("failed to publish credit balance alert",
+				"wallet_id", w.ID,
+				"error", err,
+			)
+			errs = append(errs, err)
+		}
+	}
+	if isCurrentBalanceBelowThreshold {
+		s.Logger.Infow("triggering ongoing balance alert",
+			"wallet_id", w.ID,
+			"balance", currentBalance,
+			"threshold", threshold,
+		)
+		if err := s.PublishEvent(ctx, types.WebhookEventWalletOngoingBalanceDropped, w); err != nil {
+			s.Logger.Errorw("failed to publish ongoing balance alert",
+				"wallet_id", w.ID,
+				"error", err,
+			)
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return errs[0] // Return first error
+	}
+	return nil
+}
+
+// UpdateWalletBalance updates a wallet's balance and checks thresholds
+func (s *walletService) UpdateWalletBalance(ctx context.Context, walletID string, finalBalance, newCreditBalance decimal.Decimal) error {
+	// Update balance in database
+	if err := s.WalletRepo.UpdateWalletBalance(ctx, walletID, finalBalance, newCreditBalance); err != nil {
+		return err
+	}
+
+	// Get updated wallet
+	w, err := s.WalletRepo.GetWalletByID(ctx, walletID)
+	if err != nil {
+		return err
+	}
+
+	// Get real-time balance
+	balance, err := s.GetWalletBalance(ctx, walletID)
+	if err != nil {
+		return err
+	}
+
+	// Check thresholds and trigger alerts if needed
+	return s.CheckBalanceThresholds(ctx, w, balance)
```

Comment:
UpdateWalletBalance  should do only what the name says i.e update the balance of a wallet
Also - generally not recommended to read right after you write to DB like here fetching balance right after updating in the DB (Also not very useful as we already have final balance in input)

Also there must be some function already to update the wallet balance I assume so do check